config/initializers/rack_middleware.rb

```diff
@@ -30,7 +30,9 @@
   # If using Prometheus telemetry, we want to ensure that the middleware
   # which collects and exports metrics is loaded at the start of the 
   # middleware chain to prevent any modifications to the incoming requests
-  config.middleware.insert_before(0, Monitoring::Middleware::PrometheusExporter, registry: Monitoring::Prometheus.registry, path: '/metrics')
+  if config.conjur_config.telemetry_enabled 
```

Comment:
Let's have a test case to validate that when telemetry is not enabled then a called to `GET /metrics` returns 404 (or the appropriate response).

Reply:
This behaves a bit strangely since the endpoint is exposed via middleware versus a Rails route. If the middleware isn't loaded (default behavior, metrics disabled) it will be a 401, which I believe is true for any requests that don't include an Auth header, not just /metrics.

Testing a 404 seems a bit trickier. I assume you would need to pass an Auth header to produce this, but rspec complains that there is no matching route and doesn't actually respond 404 gracefully. Plus it seems a bit weird to pass credentials for an endpoint which doesn't require them.

If metrics is enabled then the endpoint is exposed and should be expected to return 200, but this is also tricky in terms of rspec since the Prometheus components only have a chance to be initialized on app startup and presumably we can't reload the entire application for a single test.

Reply:
Glad you brought up auth. We need to exclude the metrics endpoint from authorization, see https://github.com/cyberark/conjur/blob/1540a6fb3e515b44687ca3c69a237400d284d38a/config/initializers/rack_middleware.rb#L6-L22.

Reply:
Glad you brought up auth. I think it somehow slipped through that we need to exclude the metrics endpoint from authorization, see https://github.com/cyberark/conjur/blob/1540a6fb3e515b44687ca3c69a237400d284d38a/config/initializers/rack_middleware.rb#L6-L22.

Reply:
For testing the endpoint under different configuration options we might be able to lean on cucumber which I think runs the actual rails server.